### PR TITLE
cmd/snap: remove use of zenity, use notifications for snap run inhibition

### PR DESCRIPTION
**Pull request by stolowski**
Drop zenity from snap run inhibit, use notification instead.

TODO: close the notification programatically when no longer inhibited.

### CODE REVIEW

File: cmd/snap/cmd_run_test.go
Comment (mvo5) [inline cmd/snap/cmd_run_test.go:1831]: (nitpick) this is not strictly needed, the DeepEquals in the next line should be able to deal with Nil

**Review by mvo5 — APPROVED** [review]
Thanks, this looks fine - really nice to see zenity being replaced \o/

File: cmd/snap/cmd_run_test.go
Comment (stolowski) [reply cmd/snap/cmd_run_test.go]: Indeed, thanks.

File: cmd/snap/inhibit.go
Comment (jhenstridge) [inline cmd/snap/inhibit.go:77]: Here, `userclient` will try to multiplex API calls to all available session agents on the system. In this case though, we probably want to only talk to the matching user's session agent.  Maybe we need a version of `New()` that lets us specify a subset of user IDs to contact?

While there is access control in the session agent to reject calls from other uids, I suspect we're not hitting these errors because other user's `$XDG_RUNTIME_DIR`s are not readable, so aren't picked up by the glob. It'd be better not to rely on this though.

**Review by jhenstridge — COMMENTED** [review]
The general approach seems fine, but I think we want some way to use `usersession/client` to talk to one specific session agent, rather than trying to ping every agent available on the system.

File: cmd/snap/inhibit.go
Comment (stolowski) [reply cmd/snap/inhibit.go:77]: Great point, I did notice the multiplexing aspect of doMany in the client, agree it makes sense to make it userid-based.

Comment (stolowski) [reply cmd/snap/inhibit.go:77]: I've proposed https://github.com/snapcore/snapd/pull/11072 as a prerequisite for addressing this point.

Comment (stolowski) [reply cmd/snap/inhibit.go]: Done.

**Review by mardy — APPROVED** [review]
LGTM!

**Review by bboozzoo — APPROVED** [review]
LGTM

**Review by anonymouse64 — APPROVED** [review]
lgtm

**Comment by stolowski** [comment]
This PR should get +1 from @jhenstridge before landing.

**Review by jhenstridge — APPROVED** [review]
Looks good!

**Pull request merged**
